lib/utils/addSemicolonForEditInfo.mjs

```diff
@@ -0,0 +1,30 @@
+/** @import { Node as PostcssNode } from 'postcss' */
+
+/**
+ * Adjust the text in EditInfo to include a semicolon when needed.
+ *
+ * @param {PostcssNode} node
+ * @param {{range: [number, number], text: string}} fixData
+ * @returns {{range: [number, number], text: string}}
+ */
+export default function addSemicolonForEditInfo(node, fixData) {
+	if (node.type === 'decl') {
+		if (node.parent?.raws.semicolon || node.parent?.last !== node) {
+			return {
+				...fixData,
+				text: `${fixData.text};`,
```

Comment:
question

Can the last character of `fixData.text` be `\n`?

Reply:
See: 
- https://github.com/postcss/postcss/blob/755f08f8dbdbc001906b10db3b262948f8491d82/lib/stringifier.js#L40
- https://github.com/postcss/postcss/blob/755f08f8dbdbc001906b10db3b262948f8491d82/lib/stringifier.js#L120

Adding the semicolon is always the last step in the stringifier for both at rules and declarations.

It is this step we have to manually do because the `semicolon` flag in the PostCSS stringifier is always `false` when directly calling `node.toString()`.

It is only set to `true` when serializing a parent node.